src/main/webapp/src/components/lyrics.js

```diff
@@ -20,14 +82,57 @@ class Lyrics extends React.Component {
     const lyrics = this.props.lyrics
       .replace(/\r\n/g, "\n")
       .replace(/\n\n\n/g, "\n\n");
+    
+    const speechUrl = this.state.speechUrl;
+
+    const AudioPlayerHTML = (
+      <AudioPlayer
+        src={speechUrl}
+      />);
 
-    return (
+    const LoadingProgressHTML = (
       <div>
-        <Typography variant="h4">Lyrics</Typography>
-        <p style={{ whiteSpace: "pre-wrap" }}>{lyrics}</p>
+        <p>Loading audio file...</p>
+        <StyledLinearProgress/>
       </div>
     );
+
+    const AudioErrorHTML = (
+      <p>{`Couldn't load the audio, please try again later.`}</p>
+    );
+
+    const LyricsHTML = (AudioPlayerStatus) => { return (
+      <div className={classes.root}>
+        <Typography variant="h4">Lyrics</Typography>
+        <div className={classes.audio}>
+          {AudioPlayerStatus}
+        </div>
+        <p style={{ whiteSpace: "pre-wrap" }}>{lyrics}</p>
+      </div>);};
+
+    if (this.state.error) {
+      console.log(this.state.error.message)
```

Comment:
Remove logging

Reply:
Why? I have it here so we'll know what exactly went wrong

Reply:
I guess that's OK for debugging, but keep in mind it can cause errors: https://stackoverflow.com/questions/1114187/bad-idea-to-leave-console-log-calls-in-your-production-javascript-code

Also, if it's an error, you can use [console.error](https://developer.mozilla.org/en-US/docs/Web/API/Console/error) instead.